typeset/converter.js

```diff
@@ -239,7 +243,49 @@ async function injectCoverageCollection (page) {
   }
 }
 
+const highlightCodeElements = (inputPath) => {
+  let inputFile
+  
+  try {
+    const data = fs.readFileSync(inputPath, 'utf8')
+    inputFile = data.toString()
+  } catch (err) {
+    console.error(err)
+  }
+
+  const dom = new JSDOM(inputFile, {
+    contentType: "application/xhtml+xml"
+  })
+
+  const preTagElements = [ ... dom.window.document.querySelectorAll("pre") ] 
+  preTagElements.forEach(pre => {
+    const highlightedCode = hljs.highlight('Python', pre.textContent).value  
+    pre.innerHTML = '<tempElement xmlns="http://www.w3.org/1999/xhtml">' + highlightedCode + '</tempElement>'
+    if (pre.childNodes.length !== 1) {
+      throw new Error ("Bug: Should have had one child")
+    }
+    const tempElement = pre.firstElementChild
+    tempElement.remove()
+    const children = [... tempElement.childNodes]
+    children.forEach(c => {
+      pre.append(c)
+    })
+  });
+  const highlightedOutput = serialize(dom.window.document.documentElement)
+  fs.writeFile(inputPath, highlightedOutput, (err) => {
```

Comment:
Here I'm overwriting the baked file - the same file that's being passed to `highlightCodeElements`. 
Do you think this is okay or would it be better to go from `collection.baked.xhtml` -> `collection.highlighted.xhtml` -> `collection.mathified.xhtml` ? @philschatz

Reply:
I think it is useful for future debugging to be able to re-run highlight code without having to remember to squirrel away the input file beforehand.